Mathlib/Analysis/NormedSpace/UnitizationL1.lean

```diff
@@ -0,0 +1,130 @@
+  NormedAddCommGroup.induced (WithLp 1 (Unitization 𝕜 A)) (WithLp 1 (𝕜 × A))
+    (unitization_addEquiv_prod 𝕜 A) (AddEquiv.injective _)
+
+/-- Bundle `WithLp.unitization_addEquiv_prod` as a `UniformEquiv`. -/
+noncomputable def uniformEquiv_unitization_addEquiv_prod :
+    WithLp 1 (Unitization 𝕜 A) ≃ᵤ WithLp 1 (𝕜 × A) :=
+  { unitization_addEquiv_prod 𝕜 A with
+    uniformContinuous_invFun := uniformContinuous_comap' uniformContinuous_id
+    uniformContinuous_toFun := uniformContinuous_iff.mpr le_rfl }
+
+instance instCompleteSpace [CompleteSpace 𝕜] [CompleteSpace A] :
+    CompleteSpace (WithLp 1 (Unitization 𝕜 A)) :=
+  completeSpace_congr (uniformEquiv_unitization_addEquiv_prod 𝕜 A).uniformEmbedding |>.mpr
+    CompleteSpace.prod
+
+variable {𝕜 A}
+
+open ENNReal in
+lemma unitization_norm_def (x : WithLp 1 (Unitization 𝕜 A)) :
+    ‖x‖ = ‖(WithLp.equiv 1 _ x).fst‖ + ‖(WithLp.equiv 1 _ x).snd‖ := calc
+  ‖x‖ = (‖(WithLp.equiv 1 _ x).fst‖ ^ (1 : ℝ≥0∞).toReal +
+      ‖(WithLp.equiv 1 _ x).snd‖ ^ (1 : ℝ≥0∞).toReal) ^ (1 / (1 : ℝ≥0∞).toReal) :=
+    WithLp.prod_norm_eq_add (by simp : 0 < (1 : ℝ≥0∞).toReal) _
+  _   = ‖(WithLp.equiv 1 _ x).fst‖ + ‖(WithLp.equiv 1 _ x).snd‖ := by simp
+
+lemma unitization_nnnorm_def (x : WithLp 1 (Unitization 𝕜 A)) :
+    ‖x‖₊ = ‖(WithLp.equiv 1 _ x).fst‖₊ + ‖(WithLp.equiv 1 _ x).snd‖₊ :=
+  Subtype.ext <| unitization_norm_def x
+
+lemma unitization_norm_inr (x : A) : ‖(WithLp.equiv 1 (Unitization 𝕜 A)).symm x‖ = ‖x‖ := by
+  simp [unitization_norm_def]
+
+lemma unitization_nnnorm_inr (x : A) : ‖(WithLp.equiv 1 (Unitization 𝕜 A)).symm x‖₊ = ‖x‖₊ := by
+  simp [unitization_nnnorm_def]
+
+lemma unitization_isometry_inr :
+    Isometry (fun x : A ↦ (WithLp.equiv 1 (Unitization 𝕜 A)).symm x) :=
+  AddMonoidHomClass.isometry_of_norm
+    ((WithLp.linearEquiv 1 𝕜 (Unitization 𝕜 A)).symm.comp <| Unitization.inrHom 𝕜 A)
+    unitization_norm_inr
+
+instance instUnitizationRing : Ring (WithLp 1 (Unitization 𝕜 A)) :=
+  inferInstanceAs (Ring (Unitization 𝕜 A))
+
+@[simp]
+lemma unitization_mul (x y : WithLp 1 (Unitization 𝕜 A)) :
+    WithLp.equiv 1 _ (x * y) = (WithLp.equiv 1 _ x) * (WithLp.equiv 1 _ y) :=
+  rfl
+
+instance {R : Type*} [CommSemiring R] [Algebra R 𝕜] [DistribMulAction R A] [IsScalarTower R 𝕜 A] :
+    Algebra R (WithLp 1 (Unitization 𝕜 A)) :=
+  inferInstanceAs (Algebra R (Unitization 𝕜 A))
+
+@[simp]
+lemma unitization_algebraMap (r : 𝕜) :
+    WithLp.equiv 1 _ (algebraMap 𝕜 (WithLp 1 (Unitization 𝕜 A)) r) =
+      algebraMap 𝕜 (Unitization 𝕜 A) r :=
+  rfl
+
+/-- `WithLp.equiv` bundled as an algebra isomorphism with `Unitization 𝕜 A`. -/
+@[simps!]
+def unitizationAlgEquiv (R : Type*) [CommSemiring R] [Algebra R 𝕜] [DistribMulAction R A]
+    [IsScalarTower R 𝕜 A] : WithLp 1 (Unitization 𝕜 A) ≃ₐ[R] Unitization 𝕜 A :=
+  { WithLp.equiv 1 (Unitization 𝕜 A) with
+    map_mul' := fun _ _ ↦ rfl
+    map_add' := fun _ _ ↦ rfl
+    commutes' := fun _ ↦ rfl }
```

Comment:
I think this should exist in the generality of arbitrary algebras? Is there any case where `WithLp 1 A` would be expected to have a different multiplication than `A`?

Reply:
Sure. Consider convolution on $L^1$.

Reply:
Would you expect convolution to be the *default* there though? Or would that be `WithConvolution (WithLp 1 A)`?

Reply:
Personally, I would expect it to be the default, but maybe I'm an outlier. 

My inclination would be to worry about generalizing this (the `AlgEquiv`) when we have another example where we want it.

Reply:
For what it's worth, I would be *very* surprised to see convolution as the default multiplication on `WithLp 1 A`! In any case, I think generalizing can wait until we have a use case for it.